handlers.go

```diff
@@ -139,16 +139,41 @@ func (f *faucet) authOAuthHandler(msg *apirest.APIdata, ctx *httprouter.HTTPCont
 		return ctx.Send(new(HandlerResponse).SetError(ReasonErrOauthProviderNotFound).MustMarshall(), CodeErrOauthProviderNotFound)
 	}
 
-	_, err = provider.GetOAuthToken(newRequest.Code, newRequest.RedirectURL)
+	token, err := provider.GetOAuthToken(newRequest.Code, newRequest.RedirectURL)
 	if err != nil {
 		return ctx.Send(new(HandlerResponse).SetError(ReasonErrOauthProviderError).MustMarshall(), CodeErrOauthProviderError)
 	}
 
+	profileRaw, err := provider.GetOAuthProfile(token)
+	if err != nil {
+		log.Warnw("error obtaining the profile", "err", err)
+		return ctx.Send(new(HandlerResponse).SetError(ReasonErrOauthProviderError).MustMarshall(), CodeErrOauthProviderError)
+	}
+
+	var profile map[string]interface{}
+	if err := json.Unmarshal(profileRaw, &profile); err != nil {
+		log.Warnw("error marshalling the profile", "err", err)
+		return ctx.Send(new(HandlerResponse).SetError(ReasonErrOauthProviderError).MustMarshall(), CodeErrOauthProviderError)
+	}
+
+	// Check if the oauth profile is already funded
+	fundedProfileField := profile[provider.UsernameField].(string)
+	fundedAuthType := "oauth_" + newRequest.Provider
+	if funded, t := f.storage.checkIsFunded([]byte(fundedProfileField), fundedAuthType); funded {
+		errReason := fmt.Sprintf("user %s already funded, wait until %s", fundedProfileField, t)
+		return ctx.Send(new(HandlerResponse).SetError(errReason).MustMarshall(), CodeErrFlood)
+	}
+
 	data, err := f.prepareFaucetPackage(addr, "oauth")
 	if err != nil {
 		return err
 	}
-	if err := f.storage.addFundedAddress(addr, "oauth"); err != nil {
+
+	// Add address and profile to the funded list
+	if err := f.storage.addFunded(addr.Bytes(), "oauth"); err != nil {
```

Comment:
Call it something like `addFundedUserID(userID []byte, handlerID string)` and set "oauth" as a constant.